### PR TITLE
Draft release notes for 2.0

### PR DESCRIPTION
**Pull request by AtnNn**

### CODE REVIEW

**Comment by danielmewes** [comment]
I think we should not refer to the 2.0 RC release notes, but instead have full release notes with all changes since 1.16.3.

File: NOTES.md [hunk 1]
```diff
@@ -1,3 +1,76 @@
...
+* Set `TCP_NODELAY` in all drivers (#3998)
+
+### Bug fixes ###
+
+* Ruby driver: now handle signals correctly (#4029)
+* JavaScript driver: fixed a bug that could be triggered by casting `close` twice (#4017)
+* Fetching Browserify during the build process is now more reliable (#4009)
+* Fixed a bug in `rethinkdb export` to no longer hang when certain errors occur (#4005)
+* Fallback to TCP4 when binding sockets (#4000)
+* `order_by` changes are now returned in order (#3993)
```
Comment (danielmewes) [inline NOTES.md]: This should say:
Initial values for an `order_by` changefeed are now returned in order

**Comment by mglukhovsky** [comment]
Agreed with @danielmewes -- can we include the 2.0 RC1 changes here? I'll finish reviewing when they're included.

File: NOTES.md [hunk 2]
```diff
@@ -1,3 +1,76 @@
...
+
+### New features ###
+
+* Added an asynchronous API to the Python and Ruby drivers (#2622)
+* Reached a production-ready state (#1174, #1000)
+
+### Improvements ###
+
+* Simplified the `multi_throttling` infrastructure (#4021)
+* The server now reports incantation errors to client drivers earlier (#4011)
```
Comment (danielmewes) [inline NOTES.md]: I would probably list this as a bug fix, but don't feel strongly either way.

File: NOTES.md [hunk 3]
```diff
@@ -1,3 +1,76 @@
...
+* Set `TCP_NODELAY` in all drivers (#3998)
```
Comment (danielmewes) [inline NOTES.md]: More specific:
> Set `TCP_NODELAY` in the Python and Ruby driver to avoid delays in combination with `noreply` (#3998)

**Comment by danielmewes** [comment]
Added some small notes. The new items look good.

**Comment by AtnNn** [comment]
Thanks. I've applied all your suggestions.

File: NOTES.md [hunk 4]
```diff
@@ -132,9 +131,13 @@ rethinkdb._negative_zero_check --help` for additional options.
   * Improved the handling of negative zero (#3637)
   * Correctly abort `order_by.limit` changefeeds when a table become unavailable (#3932)
   * Do not unlink files early to avoid crashing in virtual environments (#3791)
+  * Fallback to TCP4 when binding sockets (#4000)
+  * No longer crash when the data files are in a VirtualBox shared folder (#3791)
 * ReQL
   * Fixed the behavior of point changefeeds on system tables (#3944)
   * `noreplyWait` no longer waits for non-`noreply` queries (#3812)
+  * Initial values for `order_by` changefeeds are now returned in order (#3993)
+  * Reduce the size of profiles when deleting documents (#3218)
```
Comment (danielmewes) [inline NOTES.md]: For consistency: Reduce -> Reduced

File: NOTES.md [hunk 5]
```diff
@@ -1,27 +1,18 @@
...
-to test it, please [back up all][backup-docs] your data first.
-
-Please report any bugs you find on GitHub
-(http://github.com/rethinkdb/rethinkdb/issues/new) or on our mailing
-list (https://groups.google.com/forum/#!forum/rethinkdb).
+# Release 2.0.0 (Tesuji)
 
 Release highlights:
 * Support for attaching a changefeed to the `get_all` and `union` commands
 * Improved support for asynchronous queries
```
Comment (danielmewes) [inline NOTES.md]: Should we mention production readyness here?
Something like
> Fully supported, production-ready release

@mglukhovsky might have a better idea for the phrasing.

File: NOTES.md [hunk 6]
```diff
@@ -1,27 +1,18 @@
...
-Released on 2015-03-23
-
-This is a release candidate for the upcoming RethinkDB 2.0 release.
-It is not for production use. It might still have bugs. If you decide
-to test it, please [back up all][backup-docs] your data first.
-
-Please report any bugs you find on GitHub
-(http://github.com/rethinkdb/rethinkdb/issues/new) or on our mailing
-list (https://groups.google.com/forum/#!forum/rethinkdb).
+# Release 2.0.0 (Tesuji)
```
Comment (danielmewes) [inline NOTES.md]: The code name is Yojimbo

**Comment by AtnNn** [comment]
Updated, thanks.

**Comment by danielmewes** [comment]
Looks great!

File: NOTES.md [hunk 7]
```diff
@@ -72,6 +64,8 @@ rethinkdb._negative_zero_check --help` for additional options.
 
 ## New Features ##
```
Comment (mglukhovsky) [inline NOTES.md]: Downstyle capitalization:

> New features

File: NOTES.md [hunk 8]
```diff
@@ -72,6 +64,8 @@ rethinkdb._negative_zero_check --help` for additional options.
...
 
+* Overall
+  * Reached a production-ready state (#1174, #1000)
```
Comment (mglukhovsky) [inline NOTES.md]: Does this address #1000? I think there may be work left to be done. :smile:

File: NOTES.md [hunk 9]
```diff
@@ -95,6 +92,9 @@ rethinkdb._negative_zero_check --help` for additional options.
   * Execute multiple queries in parallel on a single connection (#3296)
   * Improved the performance of sending responses (#3744)
   * Immediately send back an empty first batch when the result is a changefeed (#3852)
+  * Simplified the `multi_throttling` infrastructure (#4021)
+  * The server now reports incantation errors to client drivers earlier (#4011)
```
Comment (mglukhovsky) [inline NOTES.md]: Incantation errors?

File: NOTES.md [hunk 10]
```diff
@@ -144,11 +148,44 @@ rethinkdb._negative_zero_check --help` for additional options.
   * Fixed a bug that caused autocompletion to fail in certain cases (#3143)
 * Python driver
   * Fixed `rethinkdb export` compatibility between Python 2 and Python 3 (#3911)
+  * Fixed a bug in `rethinkdb export` to no longer hang when certain errors occur (#4005)
```
Comment (mglukhovsky) [inline NOTES.md]: > Fixed a bug that caused `rethinkdb export` to hang when certain errors ocurr

File: NOTES.md [hunk 11]
```diff
@@ -144,11 +148,44 @@ rethinkdb._negative_zero_check --help` for additional options.
...
 * JavaScript driver
   * Fixed a bug that caused `cursor.each` to fail with an exception (#3826)
   * Fixed a bug that caused connection errors to be discarded (#3733)
+  * Fixed a bug that could be triggered by casting `close` twice (#4017)
+  * Fixed a bug in `feed.close` (#3967)
 * Ruby driver
   * Fixed a bug that caused failures when using JRuby (#3795)
+  * Handle signals correctly (#4029)
```
Comment (mglukhovsky) [inline NOTES.md]: > Signals are now handled correctly

**Comment by mglukhovsky** [comment]
@AtnNn -- I'm all set, pending a few comments.

**Comment by AtnNn** [comment]
Thanks. I've incorporated your suggestions.

**Pull request merged**